rule.go

```diff
@@ -164,20 +167,23 @@ func (cc *Conn) newRule(r *Rule, op ruleOperation) *Rule {
 	msgData := []byte{}
 
 	msgData = append(msgData, data...)
-	var flags netlink.HeaderFlags
 	if r.UserData != nil {
 		msgData = append(msgData, cc.marshalAttr([]netlink.Attribute{
 			{Type: unix.NFTA_RULE_USERDATA, Data: r.UserData},
 		})...)
 	}
 
+	var flags netlink.HeaderFlags
+	var handleReply func(reply netlink.Message) error
```

Comment:
Is it really required to make the reply handling a function pointer? Couldn’t we inspect the reply message flags to see if this was an echo for a create operation?

Reply:
I used a callback because that seemed like the simplest way to do it. Would you prefer the `handleReply` field of `netlinkMessage` to have an interface type instead of a function type?

Reply:
Changed it to what you suggested.